submissions/DmytryjK/task_js_oop/js/app.js

```diff
@@ -0,0 +1,164 @@
+                resetPlayerPosition(player);
+        } 
+        if (player.y <= 0) {
+            setTimeout(deletePopupMessage, 1000, 'body', 'popUp');
+            showPopupMessage('div','popUp', 'body', 'You win! :)');
+            resetPlayerPosition(player); 
+            scoreCount ++;
+            score();
+        }
+        collisionResetPositions(player, firstEnemy, secondEnemy, thirdEnemy);
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    handleInput(key) {
+        switch(key) {
+            case "left":
+                this.x += -this.heroWidth; 
+                break;
+            case "up":
+                this.y += -this.heroHeight/2; 
+                break;
+            case "right":
+                this.x += this.heroWidth; 
+                break;
+            case "down":
+                this.y += this.heroHeight/2;
+                break;
+        }
+    }
+};
+
+const   firstEnemy  = new Enemy(0, 60, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth),
+        secondEnemy  = new Enemy(0, 145, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth),
+        thirdEnemy  = new Enemy(0, 230, baseConfig.speedRandom(), baseConfig.enemyLinkImg, baseConfig.fieldWidth, baseConfig.enemyWidth);
+
+const player = new Player(baseConfig.fieldWidth, baseConfig.fieldHeight, baseConfig.playerLinkImg, baseConfig.heroWidth, baseConfig.heroHeight);
+
+const allEnemies = [firstEnemy, secondEnemy, thirdEnemy];
+
+document.addEventListener('keyup', function(e) {
+    var allowedKeys = {
+        37: 'left',
+        38: 'up',
+        39: 'right',
+        40: 'down'
+    };
+    if (!document.querySelector('body').contains(document.querySelector('.popUp'))) {
+        player.handleInput(allowedKeys[e.keyCode]);
+    }
+});
+
+function collisionResetPositions(player, firstEnemy, secondEnemy, thirdEnemy) {
+        if (player.y <= thirdEnemy.y && 
+            player.y > secondEnemy.y && 
+            player.x <= Math.floor(thirdEnemy.x) + player.heroWidth/1.5 && 
+            player.x >= Math.floor(thirdEnemy.x) - player.heroWidth/1.5 
+            ||
+            player.y <= secondEnemy.y && 
+            player.y > firstEnemy.y &&
+            player.x <= Math.floor(secondEnemy.x) + player.heroWidth/1.5 && 
+            player.x >= Math.floor(secondEnemy.x) - player.heroWidth/1.5 
+            ||
+            player.y <= firstEnemy.y && 
+            player.y > 0 && 
+            player.x <= Math.floor(firstEnemy.x) + player.heroWidth/1.5 && 
+            player.x >= Math.floor(firstEnemy.x) - player.heroWidth/1.5) {
+                setTimeout(deletePopupMessage, 1000, 'body', 'popUp');
+                showPopupMessage('div','popUp', 'body', 'You losе, bro :(');
+                resetPlayerPosition(player);
+                scoreCount = 0;
+                score(scoreCount);
+        }
+    }
```

Comment:
As a peer developer I want to add more enemies, say 100 more. There should be a way to avoid adding 5 lines of code per each enemy.

Also fix code formatting. This is important as the code is written for other humans to read.

Reply:
fixed